src/Composition/ResultElements/ContactResult.jl

```diff
@@ -0,0 +1,132 @@
+
+    object1::Object3D{F}
+    object2::Object3D{F}
+    objectCoordinateRef::Union{Object3D{F}, Nothing}
+
+    penetrationResultIndex::Int
+    penetrationVelocityResultIndex::Int
+    tangentialVelocityResultIndex::Int
+    angularVelocityResultIndex::Int
+    normalForceResultIndex::Int
+    tangentialForceResultIndex::Int
+    torqueResultIndex::Int
+    positionVectorResultIndex::Int
+    normalVectorResultIndex::Int
+    forceVectorResultIndex::Int
+    torqueVectorResultIndex::Int
+
+    function ContactResult{F}(; path::String = "",
+                                object1::Object3D{F},
+                                object2::Object3D{F},
+                                objectCoordinateRef::Union{Object3D{F}, Nothing}=nothing ) where F <: Modia3D.VarFloatType
+        return new(path, object1, object2, objectCoordinateRef)
+    end
+end
+ContactResult(; kwargs...) = ContactResult{Float64}(; kwargs...)
+
+
+function initializeResultElement(model::Modia.InstantiatedModel{F,TimeType}, result::ContactResult{F}) where {F <: Modia3D.VarFloatType, TimeType <: AbstractFloat}
+
+    result.penetrationResultIndex         = Modia.new_w_segmented_variable!(model, result.path*".penetration"        , F(0), "m")
+    result.penetrationVelocityResultIndex = Modia.new_w_segmented_variable!(model, result.path*".penetrationVelocity", F(0), "m/s")
+    result.tangentialVelocityResultIndex  = Modia.new_w_segmented_variable!(model, result.path*".tangentialVelocity" , F(0), "m/s")
+    result.angularVelocityResultIndex     = Modia.new_w_segmented_variable!(model, result.path*".angularVelocity"    , F(0), "rad/s")
+    result.normalForceResultIndex         = Modia.new_w_segmented_variable!(model, result.path*".normalForce"        , F(0), "N")
+    result.tangentialForceResultIndex     = Modia.new_w_segmented_variable!(model, result.path*".tangentialForce"    , F(0), "N")
+    result.torqueResultIndex              = Modia.new_w_segmented_variable!(model, result.path*".torque"             , F(0), "N*m")
+    result.positionVectorResultIndex      = Modia.new_w_segmented_variable!(model, result.path*".positionVector"     , SVector{3,F}(0, 0, 0), "m")
+    result.normalVectorResultIndex        = Modia.new_w_segmented_variable!(model, result.path*".normalVector"       , SVector{3,F}(0, 0, 0))
+    result.forceVectorResultIndex         = Modia.new_w_segmented_variable!(model, result.path*".forceVector"        , SVector{3,F}(0, 0, 0), "N")
+    result.torqueVectorResultIndex        = Modia.new_w_segmented_variable!(model, result.path*".torqueVector"       , SVector{3,F}(0, 0, 0), "N*m")
+
+    return nothing
+end
+
+function evaluateResultElement(model::Modia.InstantiatedModel{F,TimeType}, scene::Modia3D.Composition.Scene{F}, result::ContactResult{F}, time::TimeType) where {F <: Modia3D.VarFloatType, TimeType <: AbstractFloat}
+
+    (contactPair, converse) = getElasticContactPair(scene, result.object1, result.object2)
+    if !isnothing(contactPair)
```

Comment:
Did you check with `code_warntype` if the type of contactPair can be evaluated during runtime? If not, the types of `contactPair.results. ...` must be defined.

Reply:
Thanks, should be okay now.